### ISSUE TITLE
Create from Template causes JS error when apiVersion is not specified for all objects

### ISSUE THREAD

**Issue by tanskann** [openshift/origin-web-console#1385]
When creating application from template if apiVersion is not specified for every object, Create-button freezes but no error is shown.

```
https://foo.bar/console/scripts/scripts.js:3667:2
db@https://foo.bar/console/scripts/vendor.js:14378:44
https://foo.bar/console/scripts/vendor.js:14963:54
r@https://foo.bar/console/scripts/scripts.js:3665:24
f@https://foo.bar/console/scripts/scripts.js:3685:18
https://foo.bar/console/scripts/scripts.js:3691:54
i@https://foo.bar/console/scripts/vendor.js:5581:19
https://foo.bar/console/scripts/vendor.js:5589:2
$digest@https://foo.bar/console/scripts/vendor.js:5845:29
```

Not working:
```
- kind: DeploymentConfig
  metadata:
    labels:
      run: db
    name: db
  spec:
    replicas: 1
    template:
      metadata:
        labels:
          run: db
      spec:
        containers:
        - image: rhscl/postgresql-95-rhel7
          imagePullPolicy: IfNotPresent
          name: db
          env:
          - name: POSTGRESQL_USER
            value: myuser
          - name: POSTGRESQL_PASSWORD
            value: mypassword
          - name: POSTGRESQL_DATABASE
            value: mydatabase
          ports:
          - containerPort: 5432
            protocol: TCP
        dnsPolicy: ClusterFirst
        restartPolicy: Always
```

Working:
```
- kind: DeploymentConfig
  apiVersion: v1
  metadata:
    labels:
      run: db
    name: db
  spec:
    replicas: 1
    template:
      metadata:
        labels:
          run: db
      spec:
        containers:
        - image: rhscl/postgresql-95-rhel7
          imagePullPolicy: IfNotPresent
          name: db
          env:
          - name: POSTGRESQL_USER
            value: myuser
          - name: POSTGRESQL_PASSWORD
            value: mypassword
          - name: POSTGRESQL_DATABASE
            value: mydatabase
          ports:
          - containerPort: 5432
            protocol: TCP
        dnsPolicy: ClusterFirst
        restartPolicy: Always
```

**Comment by spadgett** [comment]
Fixed by #1512

**Comment by spadgett** [comment]
Actually, that fix does not correctly handle missing versions. Reopening.
